### DISCUSSION TITLE
Network Growth - Current Challenges & Proposed Solutions

### DISCUSSION THREAD

**Discussion by Stefaan-V** [filecoin-project/community#572] [General]
Disclaimer: information provided here is based on assumptions and may not align with future outcomes.

### **TL;DR** 

This posting summarizes a number of current economic headwinds facing the Filecoin Network, in order to motivate discussion on protocol changes, tooling and programs to address a challenging economic environment.

### **Problem Statement**

**Our macro goals are:** 

<img width="863" alt="Screen Shot 2022-11-28 at 2 16 58 PM" src="https://user-images.githubusercontent.com/83617805/204679988-84553a12-46c7-4e26-bd63-0ab2afc2cabe.png">

**The biggest blockers to achieving our macro goals are:**

<img width="705" alt="Screen Shot 2022-11-28 at 2 09 53 PM" src="https://user-images.githubusercontent.com/83617805/204679964-892277eb-7448-4e59-bf78-f6761f7ff898.png">

### Challenging Investment Environment

Given the current macro dynamics and the latest turbulence in crypto, now more than ever demonstrating real network utility is the highest priority of the Filecoin network. No surprise that the network of Storage Providers has demonstrated its vigilance by accelerating the focus of onboarding verified data and taking advantage of the FIL+ program. As more large customer datasets are coming online to the network, Storage Providers are continuously in need for more collateral. To create more awareness of the Filecoin network with existing and potential investors, we started an effort to provide information and education to the ecosystem. More needs to be done to address the magnitude of this problem. 
(Check out [tldr.filecoin.io](http://tldr.filecoin.io/) for more information)

### Storage Provider Sustainability 

Across all Storage Provider types, profit margins continue to be compressed. In the short-term, there is a need to enable the Storage Provider ecosystem with educational content and programs on how to store data and take advantage of the FIL+ program.
In the medium and long-term, we can assist Storage Providers by identifying new sources of income by charging for services, solutions and/or storing data. To accelerate this, there is a need to improve the user experience through better onramps, tooling and connectors into the traditional Web2 applications. Additionally, there is a need for recipes and solutions that demonstrate how to easily store and retrieve data. 

### Longer-term Participation

To build trust with users and ecosystem participants, the network needs to continually affirm its long-term commitment.  As we are all in the business of preserving data, the network has to demonstrate its permanence with empirical data. This empirical data needs to take the form of a wide range of long-term committed storage provider metrics and long-term client agreements.  

Additionally, the network is missing a key feature that would directly help the longer term stability and sustainability of the network: longer deal durations.  Currently, deal durations of 6-18 months are too short in comparison to the industry standard of ~3 years.  As an example, some academic research institutions who are potential clients won’t even consider archival storage solutions that are less than a minimum of 5 years.  Shorter-term deal durations do not provide a strong enough signal to potential customers or the market at large of the network’s commitment to long-term preservation. 

### Liquidity

Short commitment deal durations may contribute to more variability in FIL liquidity, which can impact price volatility that lead to less consistent returns for storage providers. Ultimately, instability in liquidity typically results in difficulty forecasting revenue projections and could erode existing and potential network participants' confidence in the storage provider network. 

### Block Rewards 

As providers are redeploying their FIL collateral to onboard more verified data by using Snapdeals to replace CC sectors, net new capacity onboarded can temporarily slow down. The total network capacity is currently predicted to drop below the baseline minting target as early as Q1 2023 (see chart). If that happens, the protocol algorithm will act as designed and incrementally reduce the total block rewards released in the network. This can impact the predicted ROI for providers that are depending on block rewards only.

<img width="1050" alt="Screen Shot 2022-11-29 at 8 53 26 PM" src="https://user-images.githubusercontent.com/83617805/204681301-7722732f-62f6-4a7c-b67d-50826234f5b3.png">
Note: More detailed analysis on the impacts of the Baseline Crossing Event will be posted by the Crypto Econ Labs within the next week. 
Disclaimer: Information provided here is based on assumptions and may not align with future outcomes.

### Community Action

The intent of this post is to ignite community discussion about the above problem statements and to solicit feedback. We as a community can propose gradual solutions that can mitigate the challenges that the network is facing. Some questions to think about are; where can we make the greatest impact?  What is the biggest blocker to network growth?  What would help improve the market understanding of Filecoin when fixed?  What problems are interconnected?  What is the business impact for you regarding any one one of the problem areas?  We encourage everyone in the community to contribute to the discussion.

### Our commitment

Moving forward, we will be posting proposed solutions for feedback. We will also share real time feedback coming out of the weekly community working groups (e.g. regional storage provider working groups, orbit events, investor discussions...). Any proposed solution should focus on simple and impactful approaches to addressing these headwinds on a continual basis.

**Comment by RobQuistNL** [comment]
The word "Retrieve" was only mentioned once, but that should be the main focus if you ask me :)

- High percentage of old deals (e.g. from slingshot 2.8 and before) are broken
- Retrieval via legacy markets / graphsync are buggy or don't work at all
- Retrieval is not easy to track (e.g. see why some are stalled / stuck in "WaitDealAcceptance")
- Retrieval via Boost works better, but still has the same issue as above
- Retrieval on the client side (the CLI / API) needs improvement
- HTTP retrieval is in a very early stage (and has been for a while - I think its even been abandoned for now)

**Reply by NSC-FIL** [reply]
I agree with RobQuistNL that retrieval of data should be improved and thoroughly tested at scale for reliability and stability.
Furthermore, I think an incremental or phased approach to increasing incentives toward storing useful and/or verified deals should be implemented on a faster pace starting with an FIP to increase max duration of sectors discussed here: https://github.com/filecoin-project/FIPs/discussions/475 .  Multipliers or other incentives to promote storing verified or useful data should be discussed, modeled (with tokenomics that will potentially increase FIL price), and deployed in phases and rolled back if ineffective or detrimental to the healthy growth of our network, IMO.

**Reply by brendalee** [reply]
Wanted to share that there's a lot of active work happening to improve retrieval from SPs, in fact - it's one of the top focus areas on the Bedrock team (if you're interested, you can read more about some upcoming [milestones](https://pl-strflt.notion.site/Retrieval-Milestones-5ea2405b34524ab68af50bd7e089190f)). 
For a lot of the bullets you mentioned, @RobQuistNL there is active work on Boost to resolve these issues, and while there's been less velocity in the recent month - http retrieval has not been abandoned. :)

**Comment by kernelogic** [comment]
1. Increase retrieval success rate
2. Increase FIL price (people more likely to invest something on the rise than knife catching on down trend.)
3. fip0036 alternative

**Comment by cryptowhizzard** [comment]
Incentivise retrievals over the course of the deal duration

**Comment by anorth** [comment]
We are close to supporting longer term deals. See https://github.com/filecoin-project/FIPs/discussions/475

**Comment by Porterstowell** [comment]
Over the past three weeks, the Storage Provider working group has been discussing ways to address the problem statements outlined above with a specific focus on Longer-Term Participation.  Here is a summary of the key discussion points specific to the ramifications of extending deal durations as well as the working recommendation that is open for debate, enhancement or a more viable alternative.  

**Macro Opportunity:**
Empirically prove the level of commitment of our storage provider and data client ecosystem to reinforce Filecoin’s longevity and stability.   

**Micro Opportunity**:
FIP 47 has introduced the option of easily extending deal durations beyond 18 months, which could lead to a FIP 48 dedicated to extending deal durations.  The February Network upgrade is in play if there is agreement by the community that a FIP 48 should be drafted for consideration. (Deadline is December 9th)

**Primitives:**
**_Incentives:_** Any proposed FIP should incentivize “Good behaviors” that are universally deemed beneficial to the network and supports the onboarding of verified data to the network.  

**_Fairness_**: Any proposed FIP shouldn’t overtly penalize or disadvantage any particular stakeholder group. 

**Potential Levers** (I.e. things that can be changed)

- **_Deal Duration_**:  What is the appropriate length of an extended storage deal?
- _**Multiplier:**_  What is the appropriate amount of additional incentive that should be provided to Storage Providers doing longer term deals (if any)?
- **_Existing Deal Eligibility_**:  Should existing deals have the ability to extend to the longer duration + multiplier or is this just for new deals only?
- **_Slashing:_**  Should there be a modification to the slashing penalties for those participating in longer term deals to prevent malicious behaviors?
- **_Collateral:_**  How can the collateral burden be mitigated to ensure all Storage Providers have the ability to participate in longer duration deals?

**Points of Agreement**

- **_Move in small increments but quickly_**: Consensus is that quick wins with continued iteration and innovation will be our most efficient path to the  desired end state. 
- **_Deal Term_**:  Set the goal of a 5 years deal duration based on market/client needs.
- **_Slashing_**:  Should not be initially touched, but should be monitored.
- _**Fast Follows:**_  If deal duration is extended, there is still a desire to address additional topics for the next network upgrade such as: a multiplier, collateral relief programs, extending current deals and other concepts still being explored.

**Points of Concern**

- **_Increased Collateral Requirements:_** Longer deal durations will increase the pledge required to secure the deal while we are already in a challenging lending/investment environment.
- **_Adverse Community Reactions_**:  Would storage providers immediately terminate sectors in order to take advantage of the longer duration opportunity?
- **_“Double Dipping”:_**  Storage Providers that do verified deals have to put up more collateral than just those providing committed capacity; collateral for the capacity + collateral for the verified data deal. 
- _**What Happens to Existing Deals?**_  Is there a path of extending existing verified data deals to the longer duration?

**Working Recommendation**
Extend Deal Durations to up to 5 years: New deals only to start with the inclination to design additional functions/capabilities/incentives over next 6-9 months based on community needs and priorities.

We encourage everyone in this discussion to weigh in with their opinions, alternatives, recommendations or enhancements to help progress the conversation.  Thank you.

**Reply by jennijuju** [reply]
> which could lead to a FIP 48 dedicated to extending deal durations.

a fip does not yet exist for this but we do have a discussion https://github.com/filecoin-project/FIPs/discussions/475. Fip-0048 is for something else.

**Comment by Fatman13** [comment]
I personally see problems when it comes to the 1st macro goal: Network Robustness: Desire for participation and investment.

### Problem 1: complex L1 protocol

In the past year, I have the opportunity to help on-board a new and very large SP (100Pib+ and more in the planning) onto the network. When training their in-house tech for running and maintaining the nodes, the amount of protocol knowledge they have to catch up and then to produce a ROI model for their executive is enormous. Complex L1 protocol has been severely delaying their efforts to on-board more data for months. 

IMO, to address this problem, future FIP should keep L1 as lean as possible to create an environment of "desire for participation and investment" and documentation on quick walk through of current FIPs that are affecting token economics.

### Problem 2: the falling of raw byte power

With fil price falling, more and more raw power SP has left the network meanwhile network upgrade drove up [gas](https://github.com/filecoin-project/FIPs/discussions/535) to further deter growth. While SPs are actively looking for storage clients, they have to report everything to f+ program to get the incentives, which increases their opex.

IMO, to address this problem, there are multiple angels we can cut into this.

- [ ] Move f+ off-chain (with respect to problem 1) or https://github.com/filecoin-project/FIPs/discussions/442
- [ ] Add support for store and retrieval of data on CC sectors

Without the ability for SP to charge FIAT to their clients, SP would always subject to the risk of FIL price fluctuation. Instead of living on the subsidy of f+, there is one SP that we have talked to figured out a model where it charges normal storage fees to its client, seal the data and earn FIL as a bonus. Nothing it stores is a storage deal on-chain. As retrieval infra like Saturn develops, IMHO, I don't see reasons that SP have to announce their deal on-chain if they have no business with FVM.

**Reply by jennijuju** [reply]
> i doubt thou that we will see native lotus support for non car file based storage and retrieval 

We have put out the lotus-market EOL notice  in v1.18.0 release note, and suggested lotus market users to migrate to boost, where the team focus on robust storage and retrieval . You can reach out to @brendalee - pm of boost team for feature requests

**Reply by jennijuju** [reply]
Cc @rjan90 on extendable lotus-miner request above

**Reply by anorth** [reply]
> Add support for store and retrieval of data on CC sectors
> ...
> As retrieval infra like Saturn develops, IMHO, I don't see reasons that SP have to announce their deal on-chain if they have no business with FVM.

The changes described in https://github.com/filecoin-project/FIPs/discussions/298 will bring this about. When we decouple the built-in market from the miner actor, the L1 protocol will no longer (be able to) restrict deal-making to the built-in market actor, or any market actor, or anything at all. So SPs will be able to put data in their sectors that's opaque to the protocol, and hence make off-chain deals with no overheads. Of course, this means that parties will not be able to rely on any on-chain enforcement of deal terms, etc, but that may be appropriate for some. FIL+ will still be on-chain, though, when it's invoked.

Alternative on-chain markets could also use stablecoin for payments.

This is currently scheduled for final design and implementation in the first half of 2023.

**Comment by Irmagogo** [comment]
Here is the example of different stages of the economy and different sources of subsidies illustrated in the following Figure.
![屏幕快照 2022-12-02 下午3 20 11](https://user-images.githubusercontent.com/112935251/205239717-a788d66b-2992-4356-bae4-9cca24677a21.png)
If we go to think about what we should do, we should first think about what direction we want to go.
Which stage are we currently at? Which part needs to be improved so that we can move into the next stage? Is the technical part and econ labs both following the path?

**Reply by nai2283** [reply]
Let's change raw bytes power to QAP when calculate baseline reward.
